lib/rules/no-unused-properties.js

```diff
@@ -221,8 +232,19 @@ module.exports = {
     const groups = new Set(options.groups || [GROUP_PROPERTY])
     const deepData = Boolean(options.deepData)
     const ignorePublicMembers = Boolean(options.ignorePublicMembers)
+    const unreferencedOptions = new Set(options.unreferencedOptions || [])
 
-    const propertyReferenceExtractor = definePropertyReferenceExtractor(context)
+    const propertyReferenceExtractor = definePropertyReferenceExtractor(
+      context,
+      {
+        unknownMemberAsUnreferenced: Boolean(
+          unreferencedOptions.has(UNREFERENCED_UNKNOWN_MEMBER)
+        ),
+        returnAsUnreferenced: Boolean(
+          unreferencedOptions.has(UNREFERENCED_RETURN)
+        )
```

Comment:
Are the explicit Boolean casts really needed? Doesn't `Set.has(…)` return a boolean already?

Reply:
Belt and braces, plus following the style of other options initialisation of being explicit, but yes you're right it's not necessary. I've taken it out, but note the rather opinionated Prettier rules now make it less readable imo...